### DISCUSSION TITLE
Mass/bulk change passwords

### DISCUSSION THREAD

**Discussion by Aldjinn** [keepassxreboot/keepassxc#8717] [Q&A] [answered]
Hi,
I'm a new user currently switching from KeePass to KeyPassXC. So far, so good by now. One feature I'm lacking is the possibility to do a mass or bulk change of password.

For example, I have a bunch of entries using the same password (bad idea, but they are used for internal applications, no public accounts). These entries have their password rotated every 4 weeks. KeePass provides KPScript for this kind of tasks.

`KPScript -c:EditEntry pwd-db.kdbx -pw:MasterPassword -ref-Password:"old\Password" -set-Password:"new-Password"`

Is this possible with KeyPassXC or keepassxc-cli, too?

Thanks. ☕

**Comment by phoerious** [comment]
You can use keepassxc-cli, but the logic has to be added externally. We do not support running scripts within KeePassXC.

**Comment by droidmonkey** [comment, marked as answer]
Use a reference between all the entries and change the password at the main entry. Syntax is documented here: https://keepassxc.org/docs/KeePassXC_UserGuide.html#_entry_cross_reference

**Reply by phoerious** [reply]
Probably the better option anyway.

**Reply by Aldjinn** [reply]
Thanks, this is a nice approach. I will give it a try.
